### DISCUSSION TITLE
Corrections to the Parsec library introduction

### DISCUSSION THREAD

**Discussion by adql** [kunigami/kunigami.github.io#7] [General]
For some reason, the introduction to the Parsec library has some operators missing (namely `<$>`, `<*>` and `<|>`) beginning with [the two flavors section](https://github.com/kunigami/kunigami.github.io/blob/master/blog/_posts/2014-01-21-an-introduction-to-the-parsec-library.md#the-two-flavors-of-the-parsec-library-monads-and-applicatives-functors). This makes it hard to follow, especially for a newcomer not yet familiar with `Applicative`.

Also, the result to `test expressionParser "(123+(324-456))"` has all the `T*` data types appear as `J*`: `Right (JNode (JTerminal 123) JAdd (JNode (JTerminal 324) JSubtract (JTerminal 456)))`.

Since that post is featured as a tutorial in the [Parsec readme](https://github.com/haskell/parsec),[^1] I thought it should be corrected. I already have it prepared and can submit a pull request, I only wanted to raise it here first since it's your personal blog.

Thanks for that helpful tutorial!

[^1]: More precisely, that readme links to the old WordPress post. I can also create a pull request there to update it.

**Comment by kunigami** [comment]
Thanks for the heads-up and the correction! I’m happy to merge the fix! I’ll be out for the next 10 days but can take a look after that :)
